add support for numpy

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #149288
* #148488
* #148485

Differential Revision: [D71294355](https://our.internmc.facebook.com/intern/diff/D71294355/)

tugsbayasgalan committed:
add support for numpy



Differential Revision: [D71294355](https://our.internmc.facebook.com/intern/diff/D71294355/)

[ghstack-poisoned]

zou3519 commented on test/export/test_export.py:
```diff
@@ -1267,6 +1267,27 @@ def false_fn(x):
 
+    def test_numpy(self):
+        class Foo(torch.nn.Module):
+            def forward(self, x):
+                a = x.numpy()
+                return x + x.numpy().sum()
+
```
What's the strategy for capturing .numpy()? What does the graph look like?

tugsbayasgalan replied:
I will update the test with the graph. BUt main strategy is just porting: https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/variables/tensor.py#L822

tugsbayasgalan committed:
Update on "add support for numpy"


Differential Revision: [D71294355](https://our.internmc.facebook.com/intern/diff/D71294355/)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "add support for numpy"


Differential Revision: [D71294355](https://our.internmc.facebook.com/intern/diff/D71294355/)

[ghstack-poisoned]
